File: Partner.Eligibility.WCF.VB/EligibilityService.vb
Imports Partner.Eligibility.Data
Imports Partner.Eligibility.Data.Entities

Public Class EligibilityService
	Implements IEligibilityService

	Public Function GetMemberInfo(memberId As String) As MemberInfo Implements IEligibilityService.GetMemberInfo
		Dim result As MemberInfo
		result = New MemberInfo

		Try
			Dim member As Member
			Using ctx As New MemberDatabaseContext
				member = ctx.Members.FirstOrDefault(Function(m) m.ExternalMemberId = memberId)
			End Using

			If Not member Is Nothing Then
				result.RequestStatus = "SUCCESSFUL"
				result.MemberId = member.ExternalMemberId
				result.FirstName = member.FirstName
				result.LastName = member.LastName
				result.Birthdate = member.Birthdate
				result.EligibilityStatus = member.EligibilityStatus
				result.CoverageStart = member.CoverageStart
				result.CoverageEnd = member.CoverageEnd
			Else
				result.RequestStatus = "INVALID"
				result.MemberId = memberId
				result.FirstName = Nothing
				result.LastName = Nothing
				result.Birthdate = DateTime.MinValue
				result.EligibilityStatus = Nothing
				result.CoverageStart = DateTime.MinValue
				result.CoverageEnd = DateTime.MinValue
			End If
		Catch ex As Exception
			Console.WriteLine(ex.Message)
		End Try

		' Set the content type of the outgoing response to application/json
		Return result
	End Function
End Class

File: Partner.Eligibility/EligibilityService.cs
using Partner.Eligibility.Data;
using Partner.Eligibility.Data.Entities;
using System;
using System.Linq;

namespace Partner.Eligibility.WCF
{
	public class EligibilityService : IEligibilityService
	{
		public MemberInfo GetMemberInfo(string memberId)
		{
			MemberInfo result = new MemberInfo();

			try
			{
				Member member;

				using (MemberDatabaseContext ctx = new MemberDatabaseContext())
				{
					member = ctx.Members.FirstOrDefault(m => m.ExternalMemberId == memberId);
				}

				if(member != null)
				{
					result.RequestStatus = "SUCCESSFUL";
					result.MemberId = member.ExternalMemberId;
					result.FirstName = member.FirstName;
					result.LastName = member.LastName;
					result.Birthdate = member.Birthdate;
					result.EligibilityStatus = member.EligibilityStatus;
					result.CoverageStart = member.CoverageStart;
					result.CoverageEnd = member.CoverageEnd;
				}
				else
				{
					result.RequestStatus = "INVALID";
					result.MemberId = memberId;
					result.FirstName = null;
					result.LastName = null;
					result.Birthdate = DateTime.MinValue;
					result.EligibilityStatus = null;
					result.CoverageStart = DateTime.MinValue;
					result.CoverageEnd = DateTime.MinValue;
				}
			}
			catch(Exception ex)
			{
				Console.WriteLine(ex.Message);
			}

			return result;
		}
	}
}
